docs: update the readme information DS-1066 (#56)
eduNEXT/eox-hooks: README.rst

| Open edX Release  |  Version       |
+===================+================+
|       Juniper     |   <= 1.0       |
+-------------------+----------------+
|        Koa        |    > 1.0       |
+-------------------+----------------+
|       Lilac       |    > 1.0 < 2.0 |
+-------------------+----------------+
|       Maple       |   >= 2.0 < 3.0 |
+-------------------+----------------+
|       Nutmeg      |   >= 3.0       |
+-------------------+----------------+
|       Olive       |   >= 4.0       |
+-------------------+----------------+
|       Palm        |   >= 5.0       |
+-------------------+----------------+
|       Quince      |   >= 6.0       |
+-------------------+----------------+
- **Post to Webhook URL:** Sends data to a specified URL based on the eox-hooks settings.
- **Get Extended Certificate Data:** Adds comprehensive certificate information to the webhook request.
- **Trigger Enrollments Creation:** Initiates an asynchronous task to enroll users in a list of courses.
- **Trigger Grades Assignment:** Propagates grades to a specified course component.
- **Create Enrollments for Program:** Enroll users in a predefined list of courses constituting a Course Program.

> The following changes to the plugin settings are necessary. If the release you are looking for is

Reviewer: Can we maintain an explanation of the settings as suggested here https://github.com/eduNEXT/eox-theming/pull/55/files#r1687159312

Author: I'll add this:

> The plugin is configured for the latest release (Quince). The following changes in the plugin settings should be applied to be used for previous releases.
> Those settings can be changed in `eox_hooks/settings/common.py`` or, for example, in the instance configurations.
> 
> 
> 🚨 If the release you are looking for is not listed, please note:
> 
> - If the Open edX release is compatible with the current eox-hooks version (see `Compatibility Notes <https://github.com/eduNEXT/eox-hooks?tab=readme-ov-file#compatibility-notes>`_), the default configuration is sufficient.
> - If incompatible, you can refer to the README from the relevant version tag for configuration details (e.g., `v5.1.0 README <https://github.com/eduNEXT/eox-hooks/blob/v5.1.0/README.rst>`_).

Is that okay?

Reviewer: yes

> #. Build the image and launch your platform with ``tutor local launch``.

Reviewer 2: ```suggestion
#. Build the image with ``tutor images build openedx --no-cache``
#. Launch your platform with ``tutor local launch``.
```

Reviewer 3: @BryanttV: why is the `--no-cache` flag needed?

Reviewer 2: We wouldn't need the command to build the image since `tutor dev launch` already does it, right? :thinking:

Author: Good point, @BryanttV, yes. With the tutor launch, we run the build process.

I'll resolve this conversation, but feel free to reopen it if you need to add more comments.

Ref: https://docs.tutor.edly.io/reference/cli/local.html#tutor-local-launch

> Before using eox-hooks, you need to set ``USE_EOX_HOOKS`` in ``True`` and configure ``EOX_HOOKS_DEFINITIONS`` as follows:

Reviewer 2: Where can I configure these settings?

Author: I'll add: in each tenant or site configuration that wants to use the plugin

Reviewer 2: Yes, I think it's better that way.

        {
            "action": "<action_name>",
            "config":
            {
                "url": "https://webhook.site",
            },
            "fail_silently": False,
            "module": "<path_to_module>",
        },
    }

> Where:
> - **trigger_event:** The name of the event the platform sends and eox-hooks listen, such as ``post_enrollment``. For a detailed list, visit the `trigger events examples file <docs/trigger_event_examples.rst>`_.

Reviewer 2: ```suggestion
Where:

- **trigger_event:** The name of the event the platform sends and eox-hooks listen, such as ``post_enrollment``. For a detailed list, visit the `trigger events examples file <docs/trigger_event_examples.rst>`_.
```

Author: Done

> Where:
> - **trigger_event:** The name of the event the platform sends and eox-hooks listen, such as ``post_enrollment``. For a detailed list, visit the `trigger events examples file <docs/trigger_event_examples.rst>`_.

Reviewer: ```suggestion
Where:

- **trigger_event:** The name of the event the platform sends and eox-hooks listen, such as ``post_enrollment``. For a detailed list, visit the `trigger events examples file <docs/trigger_event_examples.rst>`_.
```

- **action_name:** Specify the name of the action to execute. For example: ``custom_action``.
- **config:** Adds more information to the action.
- **fail_silently:** Defines what happens if an exception arises while executing the action. If ``False``, then the exception is raised.
- **path_to_module:** Path to the module defining the action to execute. For example: ``eox_hooks.actions``.

Example
########

.. code-block:: yaml

    EOX_HOOKS_DEFINITIONS:
    {
        "post_register":
        {

>             "action": "post_to_webhook_url",

Reviewer 2: From what I understand the action can be custom and hosted on any path, right?

However, I think it would also be useful to add the actions that are already available in `eox-hooks`, similar to how we did with triggers, what do you think?

Author: I agree.

Author: I added the information, but please feel free to reopen the conversation if necessary.

                "url": "https://webhook.site",
            },
            "fail_silently": True,
            "module": "eox_hooks.actions",
        },
    }

For more examples visit, the `hooks settings example file <docs/hooks_example.rst>`_.

Compatibility Notes
====================

+-------------------+----------------+
| Open edX Release  |  Version       |
+===================+================+
|       Juniper     |   <= 1.0       |
+-------------------+----------------+
|        Koa        |    > 1.0       |
+-------------------+----------------+
|       Lilac       |    > 1.0 < 2.0 |
+-------------------+----------------+
|       Maple       |   >= 2.0 < 3.0 |
+-------------------+----------------+
|       Nutmeg      |   >= 3.0       |
+-------------------+----------------+
|       Olive       |   >= 4.0       |
+-------------------+----------------+
|       Palm        |   >= 5.0       |
+-------------------+----------------+
|       Quince      |   >= 6.0       |
+-------------------+----------------+

The plugin is configured for the latest release (Quince). The following changes in the plugin settings should be applied to be used for previous releases.

> Those settings can be changed in `eox_hooks/settings/common.py`` or, for example, in the instance configurations.

Reviewer: ```suggestion
Those settings can be changed in `eox_hooks/settings/common.py` or, for example, in the instance configurations.
```
